homeassistant/components/bluetooth_le_tracker/device_tracker.py

```diff
@@ -20,6 +21,10 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+# Base UUID: 00000000-0000-1000-8000-00805F9B34FB
+# Battery characteristic: 0x2a19 (https://www.bluetooth.com/specifications/gatt/characteristics/)
+BATTERY_CHARACTERISTIC_UUID = UUID("00002a19-0000-1000-8000-00805f9b34fb")
+CONF_TRACK_BATTERY = "track_battery"
```

Comment:
Accesses to external devices (and protocol specifics) need to be contained in an external library, see https://developers.home-assistant.io/docs/development_checklist/ .

Reply:
Isn't pygatt already the library used to communicate with external devices? The code is using pygatt to read a single value and store it as an attribute. Should I create a library just to store a single constant?

Reply:
Fair enough, I'm not making the call here if this is acceptable or not, I just wanted to let you know how protocol-specific handling has usually been approached.

In a sense, pygatt is a low-level library that should be used by another library homeassistant is leveraging to perform is actions, much like any other bluetooth device is simply just using pygatt (or bluepy, or any other low-level lib). For example, integrations using HTTP as their base protocol for communications have been forced to wrap the protocol handling into an external library.